packages/meditrak-server/src/database/models/SurveyResponse.js

```diff
@@ -52,23 +52,14 @@ class SurveyResponseType extends DatabaseType {
 }
 
 export class SurveyResponseModel extends MaterializedViewLogDatabaseModel {
+  notifiers = [onChangeMarkAnswersChanged];
+
   get DatabaseTypeClass() {
     return SurveyResponseType;
   }
 
   isDeletableViaApi = true;
 
-  updateById(id, fieldsToUpdate) {
```

Comment:
Pulling the side effect out of `updateById` has four positive benefits:
1. Means the `markAsChanged` happens in the singleton db rather than the transacting db, so can reuse the existing pub/sub connection
2. Avoids overriding a central method of the base class
3. Makes it more robust to the entity or data time changing via mechanisms other than `updateById`
4. Side effects happen completely asynchronously to the actual update, and afterwards, and now won't be triggered if the update has some db error and doesn't actually process

Reply:
Good points, sounds great!